fitwsarah_frontend/src/views/AdminPanelPage/AdminAccounts.js

```diff
@@ -1,80 +1,87 @@
 import { Link } from 'react-router-dom';
-import { Container, Row, Col, Modal, Button } from 'react-bootstrap';
 import './AdminAccounts.css'; 
-import { FaSearch } from 'react-icons/fa';
+import Filter from "../../components/AdminPanel/Filter";
+import { useGetAccessToken } from "../../components/authentication/authUtils";
 
 
 
 function AdminAccounts() {
 
     const {
         isAuthenticated,
-        getAccessTokenSilently,
       } = useAuth0();
 
     const [accounts, setAccounts] = useState([]);
     const [accessToken, setAccessToken] = useState(null);
+    const [searchTerm, setSearchTerm] = useState([["accountid",""], ["username",""], ["email",""], ["city",""]]);
+
+    const getAccessToken = useGetAccessToken();
+
+    const labels = ["Account ID", "Username", "Email", "City"];
 
     useEffect(() => {
-        if (isAuthenticated) {
-          const getAccessToken = async () => {
-            try {
-              const token = await getAccessTokenSilently({
-                audience: configData.audience,
-                scope: configData.scope,
-              });
-              setAccessToken(token);
-            } catch (e) {
-              console.error(e.message);
-            }
-          };
-          getAccessToken();
-        }
-      }, [getAccessTokenSilently, isAuthenticated]);
+      const fetchToken = async () => {
+        const token = await getAccessToken();
+        if (token) setAccessToken(token);
+      };
+
+      fetchToken();
+    }, [getAccessToken]);
 
     useEffect(() => {
         if (accessToken) {
             getAllAccounts();
         }
-    }, [accessToken]);
-
-
-    
+    }, [accessToken, searchTerm]);
 
     const getAllAccounts = () => {
-        fetch("http://localhost:8080/api/v1/accounts", {
-            method: "GET",
-            headers: new Headers({
-                Authorization: "Bearer " + accessToken,
-                "Content-Type": "application/json"
-            })
-        })
-        .then((response) => {
-            if (!response.ok) {
-                throw new Error('Network response was not ok ' + response.statusText);
+      const params = new URLSearchParams();
+      searchTerm.forEach(term => {
+          if (term[1]) {
+              params.append(term[0], term[1]);
+          }
+      });
+  
+      fetch(`http://localhost:8080/api/v1/accounts${params.toString() && "?" + params.toString()}`, {
+          method: "GET",
+          headers: new Headers({
+              Authorization: "Bearer " + accessToken,
+              "Content-Type": "application/json"
+          })
+      })
+      .then((response) => {
+          if (!response.ok) {
+              throw new Error('Network response was not ok ' + response.statusText);
+          }
+          return response.json();
+      })
+      .then((data) => {
+          setAccounts(data);
+      })
+      .catch((error) => {
+          console.log(error);
+      });
+  };
+
+    function onInputChange(label, value) {
```

Comment:
Could you elaborate specifically on what the onInputChange function does? and how it works.

Reply:
onInputChange takes whats in the filter and send the a request to the API with query params.

Reply:
And it keep track of what is inside of the filter with the usesate so it doesn't get remove when they close it.